resources/web/docs.js

```diff
@@ -214,6 +216,51 @@ jQuery(function() {
               + '<a class="console_settings" title="Configure Console URL">&nbsp;</a>');
           div.find('a.console_settings').click(console_settings);
         });
+    function console_regex() {
+      // Port of https://github.com/elastic/elasticsearch/blob/master/buildSrc/src/main/groovy/org/elasticsearch/gradle/doc/RestTestsFromSnippetsTask.groovy#L71-L79
+      var method = '(GET|PUT|POST|HEAD|OPTIONS|DELETE)';
+      var pathAndQuery = '([^\\n]+)';
+      var badBody = 'GET|PUT|POST|HEAD|OPTIONS|DELETE|#';
+      var body = '((?:\\n(?!$badBody)[^\\n]+)+)'.replace('$badBody', badBody);
+      var nonComment = '$method\\s+$pathAndQuery$body?'.replace('$method', method)
+        .replace('$pathAndQuery', pathAndQuery).replace('$body', body);
+      var comment = '(#.+)';
+      return new RegExp('(?:$comment|$nonComment)\\n+'.replace('$comment', comment)
+        .replace('$nonComment', nonComment), 'g');
+    }
+    jQuery('#guide').on('click', 'a.copy_as_curl', function() {
+      var regex = console_regex();
+      var consoleText = jQuery(this).parent().prev().text() + '\n';
+      var curlText = '';
+      var match;
+      while (match = regex.exec(consoleText)) {
+        var comment = match[1];
+        var method = match[2];
+        var path = match[3];
+        var body = match[4];
+        if (comment) {
+          curlText += comment + '\n';
+        } else {
+          path = path.replace(/^\//, '');
+          path += path.includes('?') ? '&pretty' : '?pretty';
+          body = body.replace(/\'/g, '\'"\'"\'');
```

Comment:
I can try it! I don't use other shells but I can certainly use this as an excuse to spend a few minutes installing and trying a few.